crates/libcgroups/src/systemd/cpuset.rs

```diff
@@ -44,20 +44,28 @@ impl CpuSet {
     fn apply(
         cpu: &LinuxCpu,
         systemd_version: u32,
-        properties: &mut HashMap<&str, Box<dyn RefArg>>,
+        properties: &mut HashMap<&str, Variant>,
     ) -> Result<(), SystemdCpuSetError> {
         if systemd_version <= 243 {
             return Err(SystemdCpuSetError::OldSystemd);
         }
 
         if let Some(cpus) = cpu.cpus() {
-            let cpu_mask = to_bitmask(cpus).map_err(SystemdCpuSetError::CpusBitmask)?;
-            properties.insert(ALLOWED_CPUS, Box::new(cpu_mask));
+            let cpu_mask: Vec<_> = to_bitmask(cpus)
+                .map_err(SystemdCpuSetError::CpusBitmask)?
+                .into_iter()
+                .map(|v| v as u64)
+                .collect();
+            properties.insert(ALLOWED_CPUS, Variant::ArrayU64(cpu_mask));
         }
 
         if let Some(mems) = cpu.mems() {
-            let mems_mask = to_bitmask(mems).map_err(SystemdCpuSetError::MemoryNodesBitmask)?;
-            properties.insert(ALLOWED_NODES, Box::new(mems_mask));
+            let mems_mask: Vec<_> = to_bitmask(mems)
+                .map_err(SystemdCpuSetError::MemoryNodesBitmask)?
+                .into_iter()
+                .map(|v| v as u64)
+                .collect();
+            properties.insert(ALLOWED_NODES, Variant::ArrayU64(mems_mask));
```

Comment:
In tests as well as other uses, the original code expected u64, so changed this to type u64 as well, instead of casting there